lib/src/terminal_macos.rs

```diff
@@ -26,55 +30,39 @@ pub(crate) fn is_installed(terminal: Terminal) -> Result<bool, Error> {
         _ => return Err(Error::NotSupported),
     };
 
-
-    let found = match Command::new("osascript")
-        .arg("-e")
-        .arg(format!("id of application \"{}\"", app_name))
+    Command::new("osascript")
+        .args(["-e", format!("id of application \"{}\"", app_name)])
         .stderr(Stdio::null())
         .stdin(Stdio::null())
         .stdout(Stdio::null())
         .status()
-    {
-        Ok(status) => status.success(),
-        Err(err) => return Err(Error::IOError(err)),
-    };
-
-   return Ok(found)
 }
 
 fn open_with_app(app: &str, command: &str, env_vars: HashMap<String, String>) -> Result<(), Error> {
     let path = write_temp_script(command, env_vars)?;
-
-    match Command::new("open").arg("-a").arg(app).arg(path).spawn() {
-        Ok(_) => Ok(()),
-        Err(err) => Err(Error::IOError(err)),
-    }
+    Command::new("open")
+        .envs(env_vars)
+        .args(["-a", app, path])
+        .spawn()?;
+    Ok(())
 }
 
 fn write_temp_script(command: &str, env_vars: HashMap<String, String>) -> Result<PathBuf, Error> {
     let dir = temp_dir();
     let path = dir.join("run-in-terminal.sh");
 
-    let mut f = File::create(&path).map_err(Error::IOError)?;
+    let mut f = File::create(&path)?;
 
     let content = if command.is_empty() {
-        format!("#!/usr/bin/env sh\n\n{} exec $SHELL", stringify_env_vars(env_vars))
+        format!("#!/usr/bin/env sh\n\nexec $SHELL")
```

Comment:
Does this work for you? During my tests env variables are not being set when using `.envs()`